pkg/utils/pod/pod.go

```diff
@@ -34,6 +34,20 @@ func IsPodReady(pod *corev1.Pod) (ready bool, reason string) {
 	return false, "no conditions in pod status"
 }
 
+// GetPodCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetPodCondition(status *corev1.PodStatus, conditionType corev1.PodConditionType) (int, *corev1.PodCondition) {
```

Comment:
```suggestion
func GetPodCondition(status *corev1.PodStatus, conditionType corev1.PodConditionType) (*corev1.PodCondition, int) {
```

Do we need the index? If not we should remove it, it makes easier to use the function in a single line statement